sup3r/postprocessing/collection.py

```diff
@@ -485,6 +592,14 @@ def collect(cls, file_paths, out_file, features, n_writes=1,
             Flag to write status file once complete if running from pipeline.
         job_name : str
             Job name for status file if running from pipeline.
+        join_time : bool
+            Option to split full file list into chunks with each chunk having
+            the same temporal_chunk_index. The number of temporal chunk indices
+            will then be used as the number of writes. Assumes file_paths have
+            a suffix format _{temporal_chunk_index}_{spatial_chunk_index}.h5
+        mask_file : str
+            Path to meta mask containing coordinates to keep from the full
+            file list collected meta
```

Comment:
We should talk about this. It looks like you also have a requirement that there's a sparsified "gid" column, which i do not think will be the case. Best way to do this would be to run a KDTree with a very strict distance threshold to determine the mapping.

Reply:
Yeah that's a fair point. My talk with Evan seemed to indicate that he'd provide a meta file with gids but maybe not.

Reply:
Those gid's will be enumerated 1:N though where N is the final count, or at least that's going to be our desired final output. There's no reason the final meta data should have a record of the original placement in the raster. 

Also, is this mask_file intended to be the final meta data? If so, you should call it that (or "target meta"), and make sure that we are using the full dataframe as the final collected file output (e.g. make sure we're passing through the actual meta data and not just the lat/lon from the fwp).

Reply:
I provided him meta data with the original gids so I got the feeling that he would return a subset of these as the mask. Either way though I swapped this out for the KDTree approach since that is much more robust. I was thinking of the mask file as not necessarily the final meta, just a set of allowed coordinates - the final meta could still end up being a subset of the allowed coordinates for a general use case.

Reply:
I agree the KDTree will be more robust. I think we should take advantage of this "target_final_meta" to add in the actual meta data provided by evan. Otherwise we would have to add another step to overwrite the meta. Easier to offer that as an option here.

Reply:
I was thinking I'd just add a step to overwrite at the end, since I rely on the original gids to collect the spatial chunks.

Reply:
I suppose I could just replace the gid indexing with the kdtree though.

Reply:
Yeah you should be using the hi-res raster meta data, build a KDTree from that, and then query using the final meta data. The output array of index values will give you the gids from the high-res raster dataset that need to be collected along with where they belong in the output.